### PR TITLE
Add new helper MakeTitleInitCaps

### PR DESCRIPTION
**Pull request by mdhender**
Updates the hugo new command to allow creating the title using
an initial capital letter format. Adds a new helper function,
MakeTitleInitCaps, and a new config file option, coerceTitleFormat.
If set to "initCaps", then createMetaData calls the new helper
instead of MakeTitle.

Implements #2743

### CODE REVIEW

**Comment by mdhender** [comment]
I don't understand the Travis error. Make check works fine on my local.

```
$ make check
govendor test +local
?   	github.com/spf13/hugo	[no test files]
ok  	github.com/spf13/hugo/bufferpool	0.011s
ok  	github.com/spf13/hugo/commands	0.019s
ok  	github.com/spf13/hugo/create	0.016s
ok  	github.com/spf13/hugo/helpers	0.401s
ok  	github.com/spf13/hugo/hugofs	0.017s
ok  	github.com/spf13/hugo/hugolib	1.384s
?   	github.com/spf13/hugo/livereload	[no test files]
ok  	github.com/spf13/hugo/parser	0.008s
ok  	github.com/spf13/hugo/source	0.015s
ok  	github.com/spf13/hugo/target	0.017s
ok  	github.com/spf13/hugo/tpl	0.269s
ok  	github.com/spf13/hugo/transform	0.017s
?   	github.com/spf13/hugo/utils	[no test files]
?   	github.com/spf13/hugo/watcher	[no test files]
govendor test -race +local
?   	github.com/spf13/hugo	[no test files]
ok  	github.com/spf13/hugo/bufferpool	1.020s
ok  	github.com/spf13/hugo/commands	1.060s
ok  	github.com/spf13/hugo/create	1.042s
ok  	github.com/spf13/hugo/helpers	1.384s
ok  	github.com/spf13/hugo/hugofs	1.024s
ok  	github.com/spf13/hugo/hugolib	7.126s
?   	github.com/spf13/hugo/livereload	[no test files]
ok  	github.com/spf13/hugo/parser	1.018s
ok  	github.com/spf13/hugo/source	1.038s
ok  	github.com/spf13/hugo/target	1.043s
ok  	github.com/spf13/hugo/tpl	1.750s
ok  	github.com/spf13/hugo/transform	1.032s
?   	github.com/spf13/hugo/utils	[no test files]
?   	github.com/spf13/hugo/watcher	[no test files]
```

File: helpers/path.go [hunk 1]
```diff
@@ -93,6 +93,18 @@ func MakeTitle(inpath string) string {
 	return strings.Replace(strings.TrimSpace(inpath), "-", " ", -1)
 }
 
+// Issue #2743
```
Comment (moorereason) [inline helpers/path.go]: Put the "Issue" reference at the end of the comment block to keep go-lint happy.

Also, can https://github.com/bep/inflect `Titleize` help solve this?

**Comment by mdhender** [comment]
@moorereason , made that change you suggested on moving the issue number in the comment block but your comment disappeared when I squashed and pushed the changes back up.

You'd also asked about https://github.com/bep/inflect. bep had mentioned title case, too. I'm trying to avoid that and stick with just initial caps. I don't know if what I did is i18n friendly, so may end up digging more on this. I was hoping, though, that #2746 would address title case.

**Comment by bep** [comment]
I think this is too much bits and pieces for too little to be really worth it.

#2746 will be a better solution for most people. 

(btw: there is a `strings.Title` that does most of what the new func does)

**Comment by mdhender** [comment]
Tried to resolve the merge conflict with #2750 and dragged too much in with it. I'll take another whack at this later.

**Pull request closed without merging**